tex/chProgramming.tex

```diff
@@ -594,22 +594,24 @@ \section{Data types, first examples}
 \[ 0 + x \times (S\ 0) \]
 is a well formed expression in the language $\{0,S, +, \times\}$ of
 Peano arithmetic. In these examples, however, the usual axioms that
-formalize the respective arithmetic confer a distinctive status to
+give a meaning to the respective arithmetic confer a distinctive status to
 the symbols $\top$ and $\bot$ for booleans, and to the symbols $0$ and
 $S$ for natural numbers. More precisely, any variable-free boolean expression
 is equal to either $\top$ or $\bot$ modulo these axioms, and any
 variable-free expression in Peano arithmetic is equal either to $0$ or
-to an iterated application of the symbol $S$ to $0$. In both
+to an iterated application of the symbol $S$ to $0$.  In both
 cases, the other symbols in the signature represent functions, whose
-computational content is prescribed by the axioms.
+computational content is prescribed by the axioms.  In what follows,
+we shall call the distinctive symbols \emph{constructors}.  So the constructors
+in boolean arithmetic are \(\top\) and \(\bot\) and the constructors of
+Peano arithmetic are \(0\) and \(S\).
 
-In \Coq{}, such expressions are represented using a \emph{data type},
+In \Coq{}, such languages are represented using a \emph{data type},
 whose definition provides in a single declaration the name of the
-type, the symbols of (constants and) operations that \emph{build}
-elements of this type, plus some rules on how to compute on these
-elements. These data types are introduced by the means of an
+type and the constructors, plus some rules on how to compute on
+these constructors.  These data types are introduced by the means of an
```

Comment:
As commented elsewhere, I believe that "how to compute on these constructors" is unclear, and should rather be something like "how to define maps on these data types or prove theorems about their elements".